### PR TITLE
Implement save button onclick function

### PR DESCRIPTION
**Pull request by willy0416**
QSlider bug, not returning to original position but value is properly reset to 0.

### CODE REVIEW

File: first_assignment.py [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
     QCheckBox, QSlider, QComboBox, QWidget, 
-    QHBoxLayout, QVBoxLayout, QLineEdit
+    QHBoxLayout, QVBoxLayout, QLineEdit, QMessageBox
 )
 from PyQt5.QtCore import Qt
+import os
 
 class MainWindow(QMainWindow):
-    """Custom window meant to display a textbox, dropdown bar,
+    """Custom window meant to display a self.textbox, self.dropdown bar,
```
Comment (yrkim98) [inline first_assignment.py]: nit: you can just say textbox and dropdown bar for the comment best to leave code out

File: first_assignment.py [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
     viewed from top to bottom.
     """
     def __init__(self):
         """Window constructor."""
-        # Declare instance variables required for YAML file
-        self.text = ""
-        self.bool = None
-        self.choice = ""
-        self.slider_value = 0
+        self.dict = {}
```
Comment (yrkim98) [inline first_assignment.py]: dont save an instance of this dict- each time the user clicks save it's a separate instance. We dont need to know what happened before

File: first_assignment.py [hunk 3]
```diff
@@ -67,21 +72,58 @@ def update_slider_label(self, value):
...
+        # Ask user to choose a checkbox if none are selected
+        if not any(map(lambda x: x.isChecked(), self.options)):
+            QMessageBox.warning(self, "Cannot save!", "You must select an option.")
+            return
+        
+        # Grab text from text box
+        self.dict['text'] = self.textbox.text()
+
+        # Grab option from checkboxes
+        for i in range(len(self.options)):
```
Comment (yrkim98) [inline first_assignment.py]: use a foreach loop + enumerate function for this

```
for i, checkbox in enumerate(self.options):
   if checkbox.isChecked():
      self.dict["option"] = i + 1
      break

File: first_assignment.py [hunk 4]
```diff
@@ -67,21 +72,58 @@ def update_slider_label(self, value):
...
+        dialog.exec_()
+
+        # Reset fields to initial values
+        self.textbox.setText('')
+        for option in self.options:
+            option.setCheckState(Qt.Unchecked)
+        self.dropdown.setCurrentIndex(0)
+        self.slider.setTracking(True)
+        self.slider.setValue(0)
+        self.slider.setSliderPosition(0)
```
Comment (yrkim98) [inline first_assignment.py:122]: ignore slider not working for now

**Review by yrkim98 — APPROVED** [review]
looks good

File: first_assignment.py (hunk 1 above)
Comment (willy0416) [reply first_assignment.py]: Woops I think I accidentally typed those self's because I selected all instances of those words

**Pull request merged**